File: backend/src/main/java/lan/dk/podcastserver/manager/worker/Finder.java
package lan.dk.podcastserver.manager.worker;

import lan.dk.podcastserver.entity.Podcast;
import javax.validation.constraints.NotEmpty;

/**
 * Created by kevin on 22/02/15.
 */
public interface Finder {

    Podcast find(String url);
    Integer compatibility(@NotEmpty String url);
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/find/finders/Finder.kt
package com.github.davinkevin.podcastserver.find.finders

import com.github.davinkevin.podcastserver.find.FindPodcastInformation
import org.slf4j.LoggerFactory

private val log = LoggerFactory.getLogger(Finder::class.java)
/**
 * Created by kevin on 22/02/15.
 */
interface Finder {
    fun findPodcastInformation(url: String): FindPodcastInformation?
    fun compatibility(url: String): Int
}
